src/main/java/de/gebit/plugins/autoconfig/handlers/CommonConfigurationHandler.java

```diff
@@ -100,15 +117,37 @@ private void applyOnSaveOptions(OnSave options, Project project, List<String> up
 			// first, initialise the application state service used for code formatter options
 			Formatting optionsFormat = options.getFormat();
 			Formatting optionsOptimizeImports = options.getOptimizeImports();
-			TransientPluginStateService.getInstance().initFormatterSettings(getFileTypes(optionsFormat), getFileTypes(optionsOptimizeImports));
+			TransientPluginStateService.getInstance()
+					.initFormatterSettings(getFileTypes(optionsFormat), getFileTypes(optionsOptimizeImports));
 
-			setFormattingOptions(updatedConfigs, optionsFormat, "Code format on save", FormatOnSaveOptions.getInstance(project));
-			setFormattingOptions(updatedConfigs, optionsOptimizeImports, "Optimize imports on save", OptimizeImportsOnSaveOptions.getInstance(project));
+			setFormattingOptions(updatedConfigs, optionsFormat, "Code format on save",
+					FormatOnSaveOptions.getInstance(project));
+			setFormattingOptions(updatedConfigs, optionsOptimizeImports, "Optimize imports on save",
+					OptimizeImportsOnSaveOptions.getInstance(project));
+		}
+	}
+
+	private void applyProjectSDKOptions(ProjectSDK sdkOptions, Project project, List<String> updatedConfigs) {
+		if (sdkOptions != null) {
+			SdkType sdk = SdkType.findByName(sdkOptions.getType());
+			if (sdk != null) {
+				String projectSdk = JDKResolver.findProjectSdk(sdkOptions.getName(), project);
+				if (projectSdk != null) {
+					ProjectRootManager projectRootManager = ProjectRootManager.getInstance(project);
+					applySetting(projectSdk, projectRootManager.getProjectSdkName(), sdkName -> WriteAction.runAndWait(
+									() -> projectRootManager.setProjectSdkName(sdkName, sdkOptions.getType())), updatedConfigs,
+							"Project SDK");
+				}
+			} else {
+				Notifications.showWarning(
+						"SDK type \"" + sdkOptions.getType() + "\" has not been found for auto configuration", project);
```

Comment:
Maybe also show the SDK name in this warning since it may fail because of a wrong name.

Reply:
Method `findProjectSDK` takes care of notifications.